modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -385,146 +413,154 @@
-								target_buttons += "face"
-
-						if(target_vagina && target_vagina?.is_exposed())
-							target_buttons += "vagina"
-
-						if(target_anus && target_anus?.is_exposed())
-							target_buttons += "asshole"
-
-						if(target_penis && target_penis.is_exposed() && target_penis.sheath != "None")
-							target_buttons += "sheath"
-
-						target_buttons += "On them"
-
-						var/climax_into_choice = tgui_input_list(src, "Where on or in [target_human] do you wish to cum?", "Final frontier!", target_buttons)
-
-						if(!climax_into_choice)
-							create_cum_decal = TRUE
-							visible_message(span_userlove("[src] shoots their sticky load onto the floor!"), \
-								span_userlove("You shoot string after string of hot cum, hitting the floor!"))
-						else if(climax_into_choice == "On them")
-							create_cum_decal = TRUE
-							visible_message(span_userlove("[src] shoots their sticky load onto the [target_human]!"), \
-								span_userlove("You shoot string after string of hot cum onto [target_human]!"))
-						else if(climax_into_choice == "face")
-							visible_message(span_userlove("[src] shoots their sticky load onto [target_human]'s face!"), \
-								span_userlove("You shoot string after string of hot cum onto [target_human]'s face!"))
-						else
-							visible_message(span_userlove("[src] hilts [p_their()] cock into [target_human]'s [climax_into_choice], shooting cum into it!"), \
-								span_userlove("You hilt your cock into [target_human]'s [climax_into_choice], shooting cum into it!"))
-							to_chat(target_human, span_userlove("Your [climax_into_choice] fills with warm cum as [src] shoots [p_their()] load into it."))
-				try_lewd_autoemote("moan")
-				testicles.reagents.remove_all(testicles.reagents.total_volume * 0.6)
-				apply_status_effect(/datum/status_effect/climax)
-				apply_status_effect(/datum/status_effect/climax_cooldown)
-				if(create_cum_decal)
-					var/turf/our_turf = get_turf(src)
-					new /obj/effect/decal/cleanable/cum(our_turf)
-				return TRUE
-			if(vagina)
-				if(is_bottomless() || vagina.visibility_preference == GENITAL_ALWAYS_SHOW)
-					apply_status_effect(/datum/status_effect/climax)
-					apply_status_effect(/datum/status_effect/climax_cooldown)
-					visible_message(span_purple("[src] is cumming!"), span_purple("You are cumming!"))
-					var/turf/our_turf = get_turf(src)
-					new /obj/effect/decal/cleanable/femcum(our_turf)
+	if(has_status_effect(/datum/status_effect/climax_cooldown) || !(client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy)))
+		return FALSE
+
+	if(HAS_TRAIT(src, TRAIT_NEVERBONER) && has_status_effect(/datum/status_effect/climax_cooldown))
+		visible_message(span_purple("[src] twitches, trying to cum, but with no result."), \
+			span_purple("You can't have an orgasm!"))
+		return TRUE
+
+	switch(gender)
+		if(MALE)
+			playsound(get_turf(src), pick('modular_skyrat/modules/modular_items/lewd_items/sounds/final_m1.ogg',
```

Comment:
Those sounds should be handled via a define or something, because I kind of have the feeling that they're probably used in more than one spot.